src/libstd/sys/unix/thread.rs

```diff
@@ -129,14 +129,9 @@ impl Thread {
     }
 
     pub fn sleep(dur: Duration) {
-        if dur < Duration::zero() {
-            return Thread::yield_now()
-        }
-        let seconds = dur.num_seconds();
-        let ns = dur - Duration::seconds(seconds);
         let mut ts = libc::timespec {
```

Comment:
Unfortunately this can't be provided in liblibc because `Duration` doesn't exist by that point and due to the in-tree `liblibc` being different from the out-of-tree `liblibc` it unfortunately also wouldn't work to provide this in the standard library as well.